server/sonar-process/src/main/java/org/sonar/process/logging/LogLevelConfig.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.process.logging;
+
+import ch.qos.logback.classic.Level;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.sonar.process.ProcessId;
+
+import static java.util.Objects.requireNonNull;
+import static org.slf4j.Logger.ROOT_LOGGER_NAME;
+
+public final class LogLevelConfig {
+  private static final String SONAR_LOG_LEVEL_PROPERTY = "sonar.log.level";
+  private static final String PROCESS_NAME_PLACEHOLDER = "XXXX";
+  private static final String SONAR_PROCESS_LOG_LEVEL_PROPERTY = "sonar.log.level." + PROCESS_NAME_PLACEHOLDER;
+
+  private final Map<String, List<String>> configuredByProperties;
+  private final Map<String, Level> configuredByHardcodedLevel;
+
+  private LogLevelConfig(Builder builder) {
+    this.configuredByProperties = Collections.unmodifiableMap(builder.configuredByProperties);
+    this.configuredByHardcodedLevel = Collections.unmodifiableMap(builder.configuredByHardcodedLevel);
+  }
+
+  Map<String, List<String>> getConfiguredByProperties() {
+    return configuredByProperties;
+  }
+
+  Map<String, Level> getConfiguredByHardcodedLevel() {
+    return configuredByHardcodedLevel;
+  }
+
+  public static Builder newBuilder() {
```

Comment:
this method is useless as Builder constructor is public (which is good enough IMO as it's not an API)